java/pattern/Disjunction.java

```diff
@@ -51,8 +51,9 @@ public Disjunction(final List<T> patterns) {
     public Disjunction<Conjunction<Conjunctable>> normalise() {
         if (normalised == null) {
             final List<Conjunction<Conjunctable>> conjunctions = patterns.stream().flatMap(p -> {
-                if (p.isConjunction()) return Stream.of(new Conjunction<>(list(p.asConjunctable())));
+                if (p.isVariable()) return Stream.of(new Conjunction<>(list(p.asConjunctable())));
                 else if (p.isConjunction()) return p.asConjunction().normalise().patterns().stream();
+                else if (p.isNegation()) return Stream.of(new Conjunction<>(list(p.asNegation().normalise().asConjunctable())));
```

Comment:
not sure about this one...

Reply:
this seems to imply that a Conjunction can contain a normalised negation, which in general is also in Disjunction<Conjunction<...>> form?